index.js

```diff
@@ -76,13 +82,76 @@ if (process.platform === 'win32') {
 
 getFn = function () {
 	var fn = function (/*…msg*/) {
-		var start = '', end = '';
-		forEach(fn._cliColorData, function (mod) {
-			end = '\x1b[' + mod[1] + 'm' + end;
-			start += '\x1b[' + mod[0] + 'm';
-		}, null, true);
-		return start + join.call(arguments, ' ') + end;
+		var start = ''
+		  , end   = ''
+		  , keys  = Object.keys(fn._cliColorData)
+		  , msg   = join.call(arguments, ' ');
+
+		if (keys.length) {
+			for (var i=0; i<keys.length; i++) {
+				start += ( i !== 0 ? ';' : '' ) + fn._cliColorData[keys[i]][0];
+				end   += ( i !== 0 ? ';' : '' ) + fn._cliColorData[keys[i]][1];
```

Comment:
You mean the like 92 and 93? It basically use `keys` array generated by `Object.keys()`, and if looping index is different of `0` (first element), it will add a semicolon like separator. I did it to avoid this code (but is an option too):

```javascript
start = []; // overwrite
end = []; // overwrite

for (var i=0; i<keys.length; i++) {
    start.push(fn._cliColorData[keys[i]][0]);
    end.push(fn._cliColorData[keys[i]][1]);
}

start = '\x1b[' + start.join(';') + 'm';
end = '\x1b[' + end.join(';') + 'm';
```

I think that current version is more optimized, one that that will be very uncommon you nest styles with a lot of styles together (color+bg+bold, for instance), but is possible. So basically, how much you style your message, more work will do, but join is more slowly that string concat.

http://jsperf.com/cli-color-string-concat-vs-array-join